### PR TITLE
verify-owners: do not fail on removed OWNERS files

### PR DESCRIPTION
**Pull request by petr-muller**
Hit a bug in https://github.com/openshift/origin/pull/23415#pullrequestreview-264686376 and isolated it in https://github.com/openshift/release/pull/4461#discussion_r305750541

Previously, the plugin would label PRs that remove OWNERS file as having
an invalid OWNERS files, because it would attempt to parse a now
non-existent file in the PR. Make it ignore PR changes deleting files.
Add a test for this case (needed to extend some test helpers to allow removing files).

/cc @stevekuznetsov

### CODE REVIEW

File: prow/plugins/verify-owners/verify-owners.go [hunk 1]
```diff
@@ -129,7 +129,7 @@ func handle(ghc githubClient, gc *git.Client, roc repoownersClient, log *logrus.
 	// List modified OWNERS files.
 	var modifiedOwnersFiles []github.PullRequestChange
 	for _, change := range changes {
-		if filepath.Base(change.Filename) == ownersFileName {
+		if filepath.Base(change.Filename) == ownersFileName && change.Status != github.PullRequestFileRemoved {
```
Comment (nikhita) [inline prow/plugins/verify-owners/verify-owners.go]: I guess it'll also be better if we added this to OWNERS_ALIASES file check below as well. Rare to have that file removed, but you never know. :)

Comment (petr-muller) [reply prow/plugins/verify-owners/verify-owners.go]: :+1:, will fix it there too!

Comment (petr-muller) [reply prow/plugins/verify-owners/verify-owners.go]: I have looked into this case, and the current implementation handles the `OWNERS_ALIASES` removal quite fine (I have added a testcase that executes it): it does not attempt to parse that file if it does not exist, and the flow treats it like "aliases changed and now we have an empty set of aliases".

This behavior is IMHO worth keeping: compared to removing `OWNERS` (which never results in invalid state), removing aliases may result in something being really invalid, like if you remove an alias but continue to use it in unchanged `OWNERS` file. Currently, the plugin does not check for this case at all and implementing that check is IMO beyond the scope of this PR.

Sounds good?

Comment (nikhita) [reply prow/plugins/verify-owners/verify-owners.go]: Sgtm :+1:

**Pull request merged**